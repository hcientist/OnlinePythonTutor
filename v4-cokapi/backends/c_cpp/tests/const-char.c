int main() {
  const char* x = "abc";
  const char* const y = x;
}
