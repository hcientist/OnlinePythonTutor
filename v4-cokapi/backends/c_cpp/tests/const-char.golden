{
  "code": "int main() {\n  const char* x = \"abc\";\n  const char* const y = x;\n}", 
  "trace": [
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 2, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000100", 
              "pointer", 
              "<UNINITIALIZED>"
            ], 
            "y": [
              "C_DATA", 
              "0xFFF000108", 
              "pointer", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 2, 
          "ordered_varnames": [
            "x", 
            "y"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {
        "0x400594": [
          "C_ARRAY", 
          "0x400594", 
          [
            "C_DATA", 
            "0x400594", 
            "char", 
            "a"
          ], 
          [
            "C_DATA", 
            "0x400595", 
            "char", 
            "b"
          ], 
          [
            "C_DATA", 
            "0x400596", 
            "char", 
            "c"
          ], 
          [
            "C_DATA", 
            "0x400597", 
            "char", 
            "\\0"
          ]
        ]
      }, 
      "line": 3, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000100", 
              "pointer", 
              "0x400594"
            ], 
            "y": [
              "C_DATA", 
              "0xFFF000108", 
              "pointer", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 3, 
          "ordered_varnames": [
            "x", 
            "y"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {
        "0x400594": [
          "C_ARRAY", 
          "0x400594", 
          [
            "C_DATA", 
            "0x400594", 
            "char", 
            "a"
          ], 
          [
            "C_DATA", 
            "0x400595", 
            "char", 
            "b"
          ], 
          [
            "C_DATA", 
            "0x400596", 
            "char", 
            "c"
          ], 
          [
            "C_DATA", 
            "0x400597", 
            "char", 
            "\\0"
          ]
        ]
      }, 
      "line": 4, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000100", 
              "pointer", 
              "0x400594"
            ], 
            "y": [
              "C_DATA", 
              "0xFFF000108", 
              "pointer", 
              "0x400594"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 4, 
          "ordered_varnames": [
            "x", 
            "y"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }
  ]
}

